### PR TITLE
cleans up some giant spider code, moves verbs to action buttons

### PR DESCRIPTION
**Pull request by GDNgit**
## What Does This PR Do
Replaces a bunch of spawn() with do_after_once
Moves verbs to action buttons

## Why It's Good For The Game
Usability good, cleaner code good
Being able to use action buttons rather than verbs will more easily let players know they can actually use those said abilities. 

## Testing
Compiled and ran

## Changelog
:cl:
tweak: Giant spider verbs now use action buttons
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 1]
```diff
@@ -99,33 +122,30 @@
 
 /mob/living/simple_animal/hostile/poison/giant_spider/nurse/proc/GiveUp(C)
 	spawn(100)
-		if(busy == MOVING_TO_TARGET)
-			if(cocoon_target == C && get_dist(src,cocoon_target) > 1)
-				cocoon_target = null
-			busy = 0
-			stop_automated_movement = FALSE
+		if(cocoon_target == C && get_dist(src,cocoon_target) > 1)
```
Comment (Henri215) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm:125]:
```suggestion
		if(cocoon_target == C && get_dist(src, cocoon_target) > 1)
```

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 2]
```diff
@@ -134,43 +154,57 @@
...
+	var/T = loc
+
+	visible_message("<span class='notice'>[src] begins to secrete a sticky substance.</span>")
+	stop_automated_movement = TRUE
+	if(!do_after_once(src, 4 SECONDS, target = loc, attempt_cancel_message = "You stop spinning a web."))
+		return
+	new /obj/structure/spider/stickyweb(T)
+	stop_automated_movement = FALSE
+
+
```
Comment (Henri215) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm:206]:
```suggestion

```

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 3]
```diff
@@ -195,74 +229,65 @@
...
+				fed++
+				visible_message("<span class='danger'>\the [src] sticks a proboscis into \the [L] and sucks a viscous substance out.</span>")
 
-/mob/living/simple_animal/hostile/poison/giant_spider/nurse/verb/LayEggs()
-	set name = "Lay Eggs"
-	set category = "Spider"
-	set desc = "Lay a clutch of eggs, but you must wrap a creature for feeding first."
+				break
+			if(large_cocoon)
+				C.icon_state = pick("cocoon_large1","cocoon_large2","cocoon_large3")
```
Comment (Henri215) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm:267]:
```suggestion
				C.icon_state = pick("cocoon_large1", "cocoon_large2", "cocoon_large3")
```

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 4]
```diff
@@ -195,74 +229,65 @@
...
-						break
-					if(large_cocoon)
-						C.icon_state = pick("cocoon_large1","cocoon_large2","cocoon_large3")
-			cocoon_target = null
-			busy = 0
-			stop_automated_movement = FALSE
+		walk(src, 0)
+		if(!do_after_once(src, 5 SECONDS, target = cocoon_target, attempt_cancel_message = "You stop wrapping the [cocoon_target]."))
+			return
+		if(cocoon_target && isturf(cocoon_target.loc) && get_dist(src,cocoon_target) <= 1)
```
Comment (Henri215) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm:238]:
```suggestion
		if(cocoon_target && isturf(cocoon_target.loc) && get_dist(src, cocoon_target) <= 1)
```

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 5]
```diff
@@ -195,74 +229,65 @@
...
-						visible_message("<span class='danger'>\the [src] sticks a proboscis into \the [L] and sucks a viscous substance out.</span>")
-
-						break
-					if(large_cocoon)
-						C.icon_state = pick("cocoon_large1","cocoon_large2","cocoon_large3")
-			cocoon_target = null
-			busy = 0
-			stop_automated_movement = FALSE
+		walk(src, 0)
+		if(!do_after_once(src, 5 SECONDS, target = cocoon_target, attempt_cancel_message = "You stop wrapping the [cocoon_target]."))
```
Comment (lewcc) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm:236]:
```suggestion
		if(!do_after_once(src, 5 SECONDS, target = cocoon_target, attempt_cancel_message = "You stop wrapping [cocoon_target]."))
```

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 6]
```diff
@@ -195,74 +229,65 @@
...
+				if(istype(L, /mob/living/simple_animal/hostile/poison/giant_spider))
+					continue
+				if(L.stat != DEAD)
+					continue
+				large_cocoon = 1
+				L.loc = C
+				C.pixel_x = L.pixel_x
+				C.pixel_y = L.pixel_y
+				fed++
+				visible_message("<span class='danger'>\the [src] sticks a proboscis into \the [L] and sucks a viscous substance out.</span>")
```
Comment (lewcc) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm:263]:
```suggestion
				visible_message("<span class='danger'>[src] sticks a proboscis into [L] and sucks a viscous substance out.</span>")
```

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 7]
```diff
@@ -195,74 +229,65 @@
...
-						C.player_spiders = TRUE
-					fed--
-			busy = 0
-			stop_automated_movement = FALSE
+		return
+	visible_message("<span class='notice'>[src] begins to lay a cluster of eggs.</span>")
+	stop_automated_movement = TRUE
+	if(!do_after_once(src, 4 SECONDS, target = loc, attempt_cancel_message = "You stop laying eggs."))
+		return
+	var/obj/structure/spider/eggcluster/C = new /obj/structure/spider/eggcluster(src.loc)
```
Comment (lewcc) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm:283]:
```suggestion
	var/obj/structure/spider/eggcluster/C = new /obj/structure/spider/eggcluster(loc)
```

**Comment by SteelSlayer** [comment]
Should be good. Just needs reviews addressed.

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 8]
```diff
@@ -134,43 +154,55 @@
...
+/mob/living/simple_animal/hostile/poison/giant_spider/proc/create_web()
+	var/T = loc
+
+	visible_message("<span class='notice'>[src] begins to secrete a sticky substance.</span>")
+	stop_automated_movement = TRUE
+	if(!do_after_once(src, 4 SECONDS, target = loc, attempt_cancel_message = "You stop spinning a web."))
+		return
+	new /obj/structure/spider/stickyweb(T)
+	stop_automated_movement = FALSE
+/mob/living/simple_animal/hostile/poison/giant_spider/nurse/proc/wrap_target()
```
Comment (Sirryan2002) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm:205]:
```suggestion
	stop_automated_movement = FALSE

/mob/living/simple_animal/hostile/poison/giant_spider/nurse/proc/wrap_target()
```
spacing

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 9]
```diff
@@ -195,74 +227,65 @@
...
+			for(var/obj/machinery/M in C.loc)
+				if(!M.anchored)
+					M.loc = C
+					large_cocoon = 1
+			for(var/mob/living/L in C.loc)
+				if(istype(L, /mob/living/simple_animal/hostile/poison/giant_spider))
+					continue
+				if(L.stat != DEAD)
+					continue
+				large_cocoon = 1
```
Comment (Sirryan2002) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm:256]:
```suggestion
				large_cocoon = TRUE
```

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 10]
```diff
@@ -195,74 +227,65 @@
...
+			for(var/obj/item/I in C.loc)
+				I.loc = C
+			for(var/obj/structure/S in C.loc)
+				if(!S.anchored)
+					S.loc = C
+					large_cocoon = 1
+			for(var/obj/machinery/M in C.loc)
+				if(!M.anchored)
+					M.loc = C
+					large_cocoon = 1
```
Comment (Sirryan2002) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm:250]:
```suggestion
					large_cocoon = TRUE
```

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 11]
```diff
@@ -195,74 +227,65 @@
...
+			var/obj/structure/spider/cocoon/C = new(cocoon_target.loc)
+			var/large_cocoon = 0
+			C.pixel_x = cocoon_target.pixel_x
+			C.pixel_y = cocoon_target.pixel_y
+			for(var/obj/item/I in C.loc)
+				I.loc = C
+			for(var/obj/structure/S in C.loc)
+				if(!S.anchored)
+					S.loc = C
+					large_cocoon = 1
```
Comment (Sirryan2002) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm:246]:
```suggestion
					large_cocoon = TRUE
```

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 12]
```diff
@@ -195,74 +227,65 @@
...
-						C.icon_state = pick("cocoon_large1","cocoon_large2","cocoon_large3")
-			cocoon_target = null
-			busy = 0
-			stop_automated_movement = FALSE
+		walk(src, 0)
+		if(!do_after_once(src, 5 SECONDS, target = cocoon_target, attempt_cancel_message = "You stop wrapping [cocoon_target]."))
+			return
+		if(cocoon_target && isturf(cocoon_target.loc) && get_dist(src, cocoon_target) <= 1)
+			var/obj/structure/spider/cocoon/C = new(cocoon_target.loc)
+			var/large_cocoon = 0
```
Comment (Sirryan2002) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm:238]:
```suggestion
			var/large_cocoon = FALSE
```

**Review by Sirryan2002 — APPROVED** [review]

**Review by farie82 — APPROVED** [review]

**Pull request merged**